### DISCUSSION TITLE
xDrip high alert reverting to default sound after reboot.

### DISCUSSION THREAD

**Discussion by dagord** [NightscoutFoundation/xDrip#2465] [Q&A]
Hello.
I installed xDrip+ on a new phone (Xiaomi Redmi Note 10).
I've set a custom tone for high alert, but after each reboot it switches back to the "xDrip Default", even if in the alert list I still see the correct filename.
If I edit the alarm and save it again, the correct sound is played until the next reboot.
This happens only with High Alerts, while Low Alerts work as expected.
The issue didn't happen with my old phone.

![Screenshot_2022-11-15-16-30-28-091_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/21258057/201960171-3922d753-dc2f-4b18-9053-ed432f664a5c.jpg)

**Comment by Navid200** [comment]
After a reboot, does the alert page still show the correct file and yet plays the default sound?
Does, after a reboot, the alert page shows the default sound?

**Reply by dagord** [reply]
Yes. After reboot the alert page still shows the chosen file, but it plays the default sound, even in test mode.
In my case I have set two alarms (one for daytime and the other for nightime), and the issue affects both; but if after a reboot I edit the first one, the second one gets fixed as well.
But this happens onlt for "high alerts", while "low alerts" work as expected.
I tried to remove the entire list and create them again also, but the issue persists.

**Reply by Navid200** [reply]
@tzachi-dar It sounds like the custom sound file issue is still with us.

**Reply by Navid200** [reply]
@dagord What version of xDrip are you using?

**Reply by Navid200** [reply]
If you are using an old version of xDrip (older than the current stable release), please update: https://navid200.github.io/xDrip/docs/Updates

If you have updated and the problem persists, we will need to fix this.  So, please confirm.
In that case, try this and report as well: https://navid200.github.io/xDrip/docs/Custom-Alert-Sound.html

**Reply by dagord** [reply]
I'm using version 2022080314, which I think is the last stable.
On my old phone the same version worked correctly, and I replicated the settings via qr-code.
On my girlfriend phone (she's diabetic too), which runs the same version, there's no issue as well

**Reply by Navid200** [reply]
QR code has nothing to do with the alerts.  The alerts are transferred with the database.  But, that would not transfer custom sound files.

**Reply by jamorham** [reply]
@dagord please update to the latest nightly, re-select your custom audio - confirm that it works then reboot and check if the issue still persists. If so we can convert this to an issue for resolution.

**Reply by dagord** [reply]
Hmmm...
sorry for the silly question: it is safe to use a non-stable version? Otherwise I'd rather avoid updating.

**Reply by Navid200** [reply]
> Hmmm... sorry for the silly question: it is safe to use a non-stable version? Otherwise I'd rather avoid updating.

That's a fair question.
You can choose never to use an alpha or Nightly release.  And no one should blame you for that.  

There are always individuals who use alpha and there are always individuals who use Nightly.   That's how we can make progress in fixing reposted issues.
If no one ever used a Nightly, the alpha release would effectively have the stability of a Nightly.
Then, if no one ever used an alpha release, the beta release would effectively have the stability of a Nightly.  And you know where this is going.

So, you can wait for others to experience this to raise the issue until we can get the problem reproduced and fix it.
Or, you can try the release suggested.  If it makes anything worse, you can just install the stable release over it to go back.
You never need to uninstall xDrip to go back.

All that said, there is always a chance that something may misbehave.  So, you don't have to use a release that's not stable.

**Reply by dagord** [reply]
Don't take me wrong :)
I get your point: I'm a bit into developing as well and I know that nightly versions are there for testing. But since xDrip is literally a saving-life instrument, I'm wondering if - according to experience - a nightly version can really break a working functonality.
Anyway, I'll go with it and I'll give you a feedback; if the issue is not solved I'll revert to stable!

**Reply by Navid200** [reply]
A Nightly is tested by at least one developer before being released.  
For Nightly releases that bring in changes that we are worried about, for example a change that changed too many things, it is tested by multiple developers first before being released.

There are unintended changes that sometimes occur that no developer notices before and sometimes even after a release.  That's why we have bug reports:  starting here in these discussions and if they turn out to be bugs, they are moved here into the issues: https://navid200.github.io/xDrip/docs/Issues

 So, summing it all up, If you use the stable release, the issues are known.  If you use a Nightly, you may end up experiencing an issue that no one else has noticed or reported yet.

We will never just release an untested version as a Nightly with the idea that "well, it is a Nightly and no one should use it anyway".  No, absolutely not.  
It is just that the number of people who have tested a Nightly and the length of time a Nightly has been in use is much less than those for a stable release.

**Reply by dagord** [reply]
Your point makes perfectly sense, but since I dind't know how many tests are required to deploy a nightly build, I was pretty worried about this. So I hope you understand my point of view!

By the way, I installed the last Nightly and the issue persists, so I did further testings:
- the issue actually can be reproduced just by closing and relaunching the app: there's no need to reboot the phone
- it seems that if I choose one on the pre-installed tones, they work fine; otherwise local custom sound file won't; in my opening post I was using a sound dowloaded from MIUI Online Theme directory, the same I used on my old phone
- looking at the logs I can read bunch of messages such like: "setMediaDataSource from url failed" related to the sound file that was intended to be used; it seems xDrip can't access the local file; maybe a permission issue?

![Screenshot_2022-11-18-15-16-29-289_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/21258057/202731737-22cfd8b7-aab1-40b6-8e61-fcd8aae4412a.jpg)

**Reply by Navid200** [reply]
You can open an issue.  Please add a link to this discussion in it.

In the meantime, you can use the solution that has been working for me for the past 4 years since I started using xDrip highlighted as workaround here:
https://navid200.github.io/xDrip/docs/Custom-Alert-Sound.html
